XenAdmin/SettingsPanels/SnmpEditPage.cs

```diff
@@ -0,0 +1,278 @@
+﻿/* Copyright (c) Cloud Software Group, Inc. 
+ * 
+ * Redistribution and use in source and binary forms, 
+ * with or without modification, are permitted provided 
+ * that the following conditions are met: 
+ * 
+ * *   Redistributions of source code must retain the above 
+ *     copyright notice, this list of conditions and the 
+ *     following disclaimer. 
+ * *   Redistributions in binary form must reproduce the above 
+ *     copyright notice, this list of conditions and the 
+ *     following disclaimer in the documentation and/or other 
+ *     materials provided with the distribution. 
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND 
+ * CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, 
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF 
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR 
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING 
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF 
+ * SUCH DAMAGE.
+ */
+
+using System;
+using System.Drawing;
+using System.Windows.Forms;
+using XenAPI;
+using XenAdmin.Actions;
+using XenAdmin.Core;
+using System.Text.RegularExpressions;
+using XenAdmin.Actions.SNMP;
+
+namespace XenAdmin.SettingsPanels
+{
+    public partial class SnmpEditPage : UserControl, IEditPage
+    {
+        public string SubText => Messages.SNMP_EDIT_PAGE_TEXT;
+        public Image Image => Images.StaticImages._000_Network_h32bit_16;
+        private ToolTip _invalidParamToolTip = new ToolTip
+        {
+            IsBalloon = true,
+            ToolTipIcon = ToolTipIcon.Warning,
+        };
+        private string _invalidParamToolTipText = "";
+        private static readonly Regex RegexCommon = new Regex(@"^[a-zA-Z0-9-.\#@=:_]{6,32}$");
+        private static readonly Regex RegexEncryptTextBox = new Regex(@"^([a-zA-Z0-9-.\#@=:_]{8,32}|[*]{8})$");
+        private static bool _encryptTextBoxFlag = false;

```

Comment:
Setting to `false` when declaring the field is not needed because it is the default value anyway. Also, why is it `static`?